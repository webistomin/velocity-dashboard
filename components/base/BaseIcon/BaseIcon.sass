+b('base-icon')
  display: flex
  align-items: center
  justify-content: center
  border-radius: 50%
  position: relative

  &::before
    content: ''
    border-radius: inherit
    width: 100%
    height: 100%
    position: absolute
    top: 0
    left: 0
    background-color: currentColor
    opacity: 0.15

  +m('size')
    +v('xs')
      width: rem(32px)
      height: rem(32px)

    +v('s')
      width: rem(48px)
      height: rem(48px)

    +v('m')
      width: rem(70px)
      height: rem(70px)

  +m('color')
    +v('default')
      color: var(--color-primary)

  +e('icon')
    width: 50%
    height: 50%
