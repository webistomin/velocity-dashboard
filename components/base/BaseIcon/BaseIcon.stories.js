import { text } from '@storybook/addon-knobs';
import { BaseIcon } from './BaseIcon';

export default {
  title: 'BaseIcon',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
      previewTabs: {
        actions: {
          hidden: true,
        },
      },
    },
  },
};

export const Default = () => ({
  props: {
    size: {
      type: String,
      default: text('size', 'm'),
    },
    name: {
      type: String,
      default: text('name', 'icon-lightning'),
    },
    color: {
      type: String,
      default: text('color', 'gray'),
    },
  },
  render(h) {
    return h(BaseIcon, {
      props: {
        size: this.size,
        name: this.name,
        color: this.color,
      },
    });
  },
});
