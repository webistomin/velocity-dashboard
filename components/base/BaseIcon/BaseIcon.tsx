import Vue, { RenderContext, VNode, CreateElement, PropType } from 'vue';

import './BaseIcon.sass';

export type IBaseIconSize = 'xs' | 's' | 'm' | 'l';

export interface IBaseIconProps {
  size?: IBaseIconSize;
  name: string;
  color?: string;
}

export const BaseIcon = Vue.extend({
  functional: true,
  props: {
    size: {
      type: String as PropType<IBaseIconSize>,
      default: 's',
    },
    name: {
      type: String as PropType<IBaseIconProps['name']>,
      required: true,
      default: '',
    },
    color: {
      type: String as PropType<IBaseIconProps['color']>,
      default: '',
    },
  },
  render(_h: CreateElement, ctx: RenderContext<IBaseIconProps>): VNode {
    const { staticClass, class: cls } = ctx.data;
    const { size, color, name } = ctx.props;
    return (
      <span class={`base-icon base-icon_size_${size} base-icon_color_${color} ${staticClass || ''} ${cls || ''}`}>
        <svg-icon name={name} class='icon base-icon__icon' />
      </span>
    );
  },
});
