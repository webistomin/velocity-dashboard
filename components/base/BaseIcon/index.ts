import { BaseIcon } from './BaseIcon';

export default BaseIcon;
