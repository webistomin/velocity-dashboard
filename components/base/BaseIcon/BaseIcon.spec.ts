import { mount } from '@vue/test-utils';
import { BaseIcon } from './BaseIcon';

describe('BaseEmpty', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseIcon, {
      propsData: {
        ...propsData,
        name: 'icon-bubble',
      },
      ...options,
      stubs: ['svg-icon'],
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Set class from color prop', () => {
    const expectedResult = ['base-icon_color_green'];
    const wrapper = factory({
      color: 'green',
    });
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });

  it('Set default color class if color prop is not provided', () => {
    const expectedResult = ['base-icon_color_default'];
    const wrapper = factory();
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });

  it('Set size from size prop', () => {
    const expectedResult = ['base-icon_size_xs'];
    const wrapper = factory({
      size: 'xs',
    });
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });

  it('Set default size class if size prop is not provided', () => {
    const expectedResult = ['base-icon_size_s'];
    const wrapper = factory();
    const classes = wrapper.classes();
    expect(classes).toEqual(expect.arrayContaining(expectedResult));
  });
});
