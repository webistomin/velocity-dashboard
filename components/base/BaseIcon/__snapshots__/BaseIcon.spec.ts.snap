// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseEmpty Matches snapshot 1`] = `"<span class=\\"base-icon base-icon_size_s base-icon_color_default  \\"><svg-icon-stub name=\\"icon-bubble\\" class=\\"icon base-icon__icon\\"></svg-icon-stub></span>"`;
